### ISSUE TITLE
AsyncCache does automatic retries on failures

### ISSUE THREAD

**Issue by j82w** [Azure/azure-cosmos-dotnet-v3#3052]
The AsyncCache will automatically retry requests on failures for concurrent requests. This can cause additional latency and unnecessary load to gateway for some caches.

Scenario:
Request A -> slow request  
Request B -> waits on Task A from Request A
Task A -> Fails
Request A -> throws failure
Request B -> catches/ignore Task A failure. It creates a new Task to attempt to load the same value. 

The problem with this is Task A could take up to 1 minute to fail. Then the request B will wait another minute possibly and still fail.

https://github.com/Azure/azure-cosmos-dotnet-v3/blob/42fd34035541fd689e7a92c4b2bc9965f8d1b4f5/Microsoft.Azure.Cosmos/src/Routing/AsyncCache.cs#L112

Ideal scenario:
Request A -> slow request  
Request B -> waits on Task A from Request A
Task A -> Fails
Request A -> throws failure
Request B -> throws failure

Then Request A and Request B can both fail. Then each retry policy for those requests can decide to retry if necessary. 

```c#
private int TaskStartedCount = 0;
        private int GetIntCount = 0;
        private bool DelayTasks = true;

        [TestMethod]
        public async Task LazyAsync()
        {
            AsyncCache<string, bool> cache = new AsyncCache<string, bool>();
            
            List<Task> tasks = new List<Task>();
            for(int i = 0; i < 2; i++)
            {
                tasks.Add(this.CreateValue(cache));
            }

            Stopwatch sw = Stopwatch.StartNew();
            while (this.TaskStartedCount < tasks.Count && sw.Elapsed.TotalSeconds < 2)
            {
                await Task.Delay(TimeSpan.FromMilliseconds(50));
            }

            Assert.AreEqual(tasks.Count, this.TaskStartedCount, "Tasks did not start");
            this.DelayTasks = false;

            await Task.WhenAll(tasks);

            Assert.AreEqual(1, this.GetIntCount);
        }

        private async Task<bool> CreateValue(AsyncCache<string, bool> cache)
        {
            Interlocked.Increment(ref this.TaskStartedCount);
            try
            {
                return await cache.GetAsync(
                   "test",
                   default,
                   this.GetIntAsync,
                   default,
                   forceRefresh: false);
            }
            catch(ArgumentException)
            {
                return false;
            }
        }

        private async Task<bool> GetIntAsync()
        {
            Interlocked.Increment(ref this.GetIntCount);
            while (this.DelayTasks)
            {
                await Task.Delay(TimeSpan.FromMilliseconds(100));
            } 

            throw new ArgumentException("Test exception");
        }
```

**Comment by j82w** [comment]
Closing because almost all SDK caches have moved to nonblocking implementation which does not have this issue.
